modules/core/src/lib/layer-manager.js

```diff
@@ -372,6 +374,14 @@ export default class LayerManager {
     // Set back pointer (used in picking)
     layer.internalState.layer = layer;
 
+    layer.props.extensions.forEach(extension => {
+      const name = extension.constructor.name;
+      if (!this._registeredExtensions[name]) {
+        extension.addShaderHooks(this.context.programManager);
+        this._registeredExtensions[name] = true;
+      }
+    });
```

Comment:
@Pessimistress this is the piece I'd most like to get your opinion on. It's what allows an extension to add it's injections to a program manager.

Reply:
We used to support an `inject` field in the shader module definition and let luma's `assembleShaders` handle it. From deck's perspective that makes more sense - all injections are defined with modules, and modules should not be aware of the context being used.

Reply:
Done.